table/block.cc

```diff
@@ -223,6 +224,120 @@ void DataBlockIter::Seek(const Slice& target) {
   }
 }
 
+// Optimized Seek for point lookup for an internal key `target`
+// target = "seek_user_key @ seqno".
+//
+// If the return value is FALSE, iter location is undefined, and it means:
+// 1) there is no key in this block falling into the range:
+//    ["seek_user_key @ seqno", "seek_user_key @ 0"], inclusive; AND
+// 2) the last key of this block has a greater user_key from seek_user_key
```

Comment:
This definition is great. But be careful that an internal key not just has user key and sequence number, but also the type. You may want to mention this to be strictly correct.